### ISSUE TITLE
[HyperV] Allow user to input license key (BYOL) on deployment/reinstallation.

### ISSUE THREAD

**Issue by FeEzioXIII** [FuseCP/SolidCP#82]
**Description:**
Allow user to supply their own key for Windows installation in case they want to install specific keys that's not activated by AVMA.

**Ideas:**
Currently, SolidCP already had params in unattended files to change timezone, computer name, ... We can create a field in OStemplate such as: AVMA/VL key, if it's not provided, we will prompt user to supply their own license key on deployment/reinstallation.

**Comment by FuseCP-BKruszewski** [comment]
This feature should be optional, so the serveradmin should be able to en-/disable it.
If someone has licensed his hosting hardware with SPLA it's not allowed to use other licenses, but only SPLA, as far as I know.

**Comment by FeEzioXIII** [comment]
> This feature should be optional, so the serveradmin should be able to en-/disable it.
> If someone has licensed his hosting hardware with SPLA it's not allowed to use other licenses, but only SPLA, as far as I know.

Yes, that's why I mentioned if there is a key entered on template (which is SPLA/VL/AVMA key), users won't be prompted for license. However, if there is no license entered, user will be required to enter license in order to create/deploy the chosen template.

SolidCP is not only limited to hosting provider too! So it's nice that we can cover all the use cases.

**Comment by berkut1** [comment]
I think this can be done using a quota, just add a flag for third-party license keys, and when selecting the operating system, if the quota is enabled, show a field to enter the key.
